File: ApiVirtualTienda/Entity/Producto.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entity
{
    public class Producto
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public string Codigo { get; set; }
        public string Descripcion { get; set; }
        public int Cantidad { get; set; }
        public DateTime Fecha { get; set; }
        public string Estado { get; set; }
        [Column(TypeName = "decimal(17,4)")]
        public decimal Descuento { get; set; }
        [NotMapped]
        public Proveedor Proveedor { get; set; }
        [ForeignKey("ProveedorNIT")]
        public string ProveedorNIT { get; set; }
        [Column(TypeName = "decimal(17,4)")]
        public decimal IVA { get; set; }
        [Column(TypeName = "decimal(17,4)")]
        public decimal ValorDescontado { get; set; }
        [Column(TypeName = "decimal(17,4)")]
        public decimal ValorUnitario { get; set; }

        public Producto()
        {
            
        }


        
    }
}

File: FrontEndVirtualTienda/src/app/models/producto.ts
import { Proveedor } from "./proveedor";

export class Producto {
    codigo: string | undefined;
    descripcion: string;
    cantidad: number;
    fecha: Date;
    descuento: number;
    iva: number;
    valorDescontado: number;
    valorTotal: number;
    proveedor: Proveedor;
    idProveedor: string;
    valorUnitario: number;
    cantidadSeleccionada: number = 0;

    constructor(){
        this.codigo = undefined;
        this.descripcion = '';
        this.cantidad = 0;
        this.fecha = new Date();
        this.descuento = 0;
        this.iva = 0;
        this.valorDescontado = 0;
        this.valorTotal = 0;
        this.proveedor = new Proveedor();
        this.idProveedor = '';
        this.valorUnitario = 0;
    }
}
